src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -105,18 +141,36 @@ private void setupGitLabCIConfiguration(VcsRepositoryUrl repositoryURL, Programm
         }
 
         try {
-            // TODO: Update variables
-            projectApi.createVariable(repositoryPath, VARIABLE_DOCKER_IMAGE_NAME,
-                    programmingLanguageConfiguration.getImage(exercise.getProgrammingLanguage(), Optional.ofNullable(exercise.getProjectType())), Variable.Type.ENV_VAR, false,
-                    true);
-            projectApi.createVariable(repositoryPath, VARIABLE_BRANCH_NAME, exercise.getBranch(), Variable.Type.ENV_VAR, false, true);
-            projectApi.createVariable(repositoryPath, VARIABLE_ARTEMIS_SERVER_URL_NAME, artemisServerUrl.toExternalForm(), Variable.Type.ENV_VAR, false, true);
+            // TODO: Reduce the number of API calls
+
+            updateVariable(repositoryPath, VARIABLE_BUILD_DOCKER_IMAGE_NAME,
+                    programmingLanguageConfiguration.getImage(exercise.getProgrammingLanguage(), Optional.ofNullable(exercise.getProjectType())), true);
+            updateVariable(repositoryPath, VARIABLE_BUILD_LOGS_FILE_NAME, "build.log", true);
+            updateVariable(repositoryPath, VARIABLE_BUILD_PLAN_ID_NAME, buildPlanId, true);
+            updateVariable(repositoryPath, VARIABLE_CUSTOM_FEEDBACK_DIR_NAME, "TODO", false); // TODO
+            updateVariable(repositoryPath, VARIABLE_NOTIFICATION_PLUGIN_DOCKER_IMAGE_NAME, notificationPluginDockerImage, true);
+            updateVariable(repositoryPath, VARIABLE_NOTIFICATION_SECRET_NAME, artemisAuthenticationTokenValue, true);
+            updateVariable(repositoryPath, VARIABLE_NOTIFICATION_URL_NAME, artemisServerUrl.toExternalForm() + Constants.NEW_RESULT_RESOURCE_API_PATH, false);
+            updateVariable(repositoryPath, VARIABLE_SUBMISSION_GIT_BRANCH_NAME, exercise.getBranch(), false);
+            updateVariable(repositoryPath, VARIABLE_TEST_GIT_BRANCH_NAME, exercise.getBranch(), false);
+            updateVariable(repositoryPath, VARIABLE_TEST_GIT_REPOSITORY_SLUG_NAME, urlService.getRepositorySlugFromRepositoryUrlString(exercise.getTestRepositoryUrl()), true);
+            // TODO: Use a token that is only valid for the test repository for each programming exercise
+            updateVariable(repositoryPath, VARIABLE_TEST_GIT_TOKEN, gitlabToken, true);
+            // We do not mask this variable because GitLab requires that a masked string to be at least 8 characters long. This is not the case if you use the default user
+            // ('root').
+            updateVariable(repositoryPath, VARIABLE_TEST_GIT_USER, gitlabUser, false);
```

Comment:
So we could replace the `false` with something like `gitlabUser.length() >= 8`?

Reply:
I have added a regex, which covers all the requirements for masking variables.